drivers/auxiliary/cheapodc.h

```diff
@@ -104,41 +116,66 @@ class CheapoDC : public INDI::DefaultDevice
 private:
     enum controllerMode
     {
-        AUTOMATIC,
+        AUTOMATIC = 0,
         MANUAL,
         OFF,
     };
 
     enum temperatureMode
     {
-        WEATHER_QUERY,
+        WEATHER_QUERY = 0,
         EXTERNAL_INPUT
     };
 
     enum setPointMode
     {
-        DEWPOINT,
+        DEWPOINT = 0,
         TEMPERATURE,
         MIDPOINT
     };
 
+    enum weatherSource
+    {
+        OPENMETEO = 0,
+        OPENWEATHER,
+        EXTERNALSOURCE
+    };
+
+    enum CheapoDCLocation
+        {
+            LOCATION_LATITUDE,
+            LOCATION_LONGITUDE
+        };
+
+    bool fwVOneDetected = false;
     int timerIndex;
     unsigned int previousControllerMode = MANUAL;
     unsigned int prevMinOutput = 0;
     unsigned int prevMaxOutput = 100;
-    int snoopLocationIndex = 1; // Default is Disabled
-    int snoopTemperatureIndex = 1; // Default is Disabled
-    int prevSnoopLocationIndex = snoopLocationIndex;
+    int snoopLocationIndex = INDI_ENABLED; // Default is Enabled
+    int snoopTemperatureIndex = INDI_DISABLED; // Default is Disabled
+    int previousSnoopTemperatureIndex = snoopTemperatureIndex; 
+    unsigned int previousTemperatureMode = WEATHER_QUERY;
+    int snoopWeatherIndex = INDI_DISABLED; // Default is Disabled
+    int previousSnoopWeatherIndex = snoopWeatherIndex;
     char locationDevice[MAXINDIDEVICE] = {"Telescope Simulator"};
     char locationProperty[MAXINDINAME] = {"GEOGRAPHIC_COORD"};
     char locationLatAttribute[MAXINDINAME] = {"LAT"};
     char locationLongAttribute[MAXINDINAME] = {"LONG"};
     char temperatureDevice[MAXINDIDEVICE] = {"Focuser Simulator"};
     char temperatureProperty[MAXINDINAME] = {"FOCUS_TEMPERATURE"};
     char temperatureAttribute[MAXINDINAME] = {"TEMPERATURE"};
```

Comment:
What do you need these for? GEOGRAPHIC_COORD and FOCUS_TEMPERATURE are both **standard** INDI properties. If you want to snoop other devices, you need to have a single property [ActiveDeviceTP](https://github.com/indilib/indi/blob/9d6e87abfddec3d79a4ddf34b0b693f542eb6fb8/libs/indibase/indiccd.cpp#L491) for all the different types of devices you need to snoop.

If you check the example above, you'll see how INDI::CCD snoops the properties from other devices.

Reply:
Looking back likely not understanding how ActiveDeviceTP works. I will look at it.

Thanks Jasem.

Reply:
Looking at INDI:CCD, I see that it uses IUGetConfigText to fill the intial text for the ActiveDeviceTP property with any values that may be  found in the config file. This is similar to what I intended above. Moving to using only Property.load(); removes this function.  Is there a better or preferred way to do this?

https://github.com/indilib/indi/blob/9d6e87abfddec3d79a4ddf34b0b693f542eb6fb8/libs/indibase/indiccd.cpp#L479C1-L498C1

Reply:
You're right, this code is outdated and need to be removed.

Reply:
INDI::CCD code is updated.

Reply:
Updated to using ActiveDeviceTP property.

Will update the driver listing on indilib.org on Monday.